tools/serverpod_cli/bin/create/create.dart

```diff
@@ -19,12 +19,25 @@ const _defaultPorts = <String, int>{
   'Redis server': 8091,
 };
 
+bool isNameValid(String name) {
+  RegExp namePattern = new RegExp(r'^[0-9a-z]+(?:[0-9a-z]+|_[0-9a-z]+)*$');
+
+  return namePattern.stringMatch(name).toString().length == name.length;
```

Comment:
Please don't compare the string length. If you do so, the same as I previously mentioned 
https://github.com/serverpod/serverpod/pull/455#discussion_r1043863281 will be able to replicate.
I will explain to you why and what is happening. 
So, I gave a name of length 4 `!%&#`. So as the code mentioned above
```dart
namePattern.stringMatch(name);
```
will return you null cause nothing is matching the regexp. And when you stringify the _`null`_, it will return you **`null`**. Also, it has a length of 4. And as you compare the length size, it will match and return true. So the test cases fail.

Reply:
<img width="1147" alt="Screenshot 2022-12-19 at 6 00 23 PM" src="https://user-images.githubusercontent.com/35523357/208427734-6b1363b7-1b4b-4cb6-b30b-2989602d77d4.png">

Reply:
Also please write a test case.

Reply:
I have never ever written a test Case in dart. so if you could kindly tell me where to create the test within the "tests" folder. shall I create a new folder?

Reply:
Btw done with the Regex part I guess.. :)